src/Main.cpp

```diff
@@ -1,10 +1,466 @@
+using namespace deep_trekker;
+using namespace std;
+using namespace rtc;
+using std::shared_ptr;
+using std::weak_ptr;
+
+template <class T> weak_ptr<T> make_weak_ptr(shared_ptr<T> ptr) { return ptr; }
+void pong(
+    shared_ptr<rtc::WebSocket>& websocket,
+    RustySignalMessageDecoder& decoder,
+    string const& local_peer_id
+);
+void sendInitialPayload(shared_ptr<rtc::WebSocket>& websocket);
+void sendSessionCheck(shared_ptr<rtc::WebSocket>& websocket, string& local_peer_id);
+void joinSession(
+    shared_ptr<rtc::WebSocket>& websocket,
+    SignalRMessageDecoder& decoder,
+    string& local_peer_id
+);
+void leaveSession(
+    shared_ptr<rtc::WebSocket>& websocket,
+    SignalRMessageDecoder& decoder,
+    string& local_peer_id
+);
+bool initialHandShakeFinalized(
+    shared_ptr<rtc::WebSocket>& websocket,
+    SignalRMessageDecoder& decoder,
+    string& local_peer_id
+);
+void offerAnswerRustyMessageParser(
+    shared_ptr<rtc::WebSocket>& websocket,
+    RustySignalMessageDecoder& decoder
+);
+void candidateRustyMessageParser(
+    shared_ptr<rtc::WebSocket>& websocket,
+    RustySignalMessageDecoder& decoder
+);
+void offerAnswerSignalMessageParser(
+    shared_ptr<rtc::WebSocket>& websocket,
+    SignalRMessageDecoder& decoder
+);
+void candidateSignalMessageParser(
+    shared_ptr<rtc::WebSocket>& websocket,
+    SignalRMessageDecoder& decoder
+);
 
 int main(int argc, char** argv)
+try
 {
-    dummy_project::DummyClass dummyClass;
-    dummyClass.welcome();
+    if (argc != 4)
+    {
+        cout << "usage: " << argv[0] << " local_peer_id rusty_signal_server_host stun_server" << endl;
+        exit(1);
+    }
+    string local_peer_id = argv[1];
+    string rusty_signal_server_host = argv[2];
+    string stun_server = argv[3];
+
+    // Http post request
+    curlpp::Cleanup cleaner;
+    curlpp::Easy request;
+    string url_https =
+        "https://" + local_peer_id + ":5001/sessionHub/negotiate?negotiateVersion=1";
+    request.setOpt(new curlpp::options::Url(url_https));
+    request.setOpt(new curlpp::options::Verbose(true));
+    list<string> header;
+    header.push_back("Content-Type: application/x-www-form-urlencoded");
+    request.setOpt(new curlpp::options::HttpHeader(header));
+    request.setOpt(new curlpp::options::SslVerifyPeer(false));
+    std::ostringstream os;
+    curlpp::options::WriteStream ws(&os);
+    request.setOpt(ws);
+    request.perform();
+    os << request;
+    HttpPostRequestReplyMessageDecoder decoder = HttpPostRequestReplyMessageDecoder();
+    string error;
+    if (!decoder.parseJSONMessage(os.str().c_str(), error))
+    {
+        throw invalid_argument(error);
+    }
+    string https_token = decoder.getConnectionToken();
+
+    SignalRMessageDecoder signalr_decoder = SignalRMessageDecoder();
+    promise<void> wsr_promise;
+    future<void> wsr_future = wsr_promise.get_future();
+    rtc::Configuration signalr_config;
+    signalr_config.iceServers.emplace_back("stun:" + stun_server);
```

Comment:
Clean some of this up, it is a bit hard to read. Split some commands in groups separated by spaces.